bridges/snowbridge/parachain/pallets/system/src/tests.rs

```diff
@@ -0,0 +1,639 @@
+			BadOrigin
+		);
+
+		// None origin not allowed
+		assert_noop!(EthereumSystem::create_agent(RuntimeOrigin::none()), BadOrigin);
+	});
+}
+
+#[test]
+fn upgrade_as_root() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let address: H160 = Default::default();
+		let code_hash: H256 = Default::default();
+
+		assert_ok!(EthereumSystem::upgrade(origin, address, code_hash, None));
+
+		System::assert_last_event(RuntimeEvent::EthereumSystem(crate::Event::Upgrade {
+			impl_address: address,
+			impl_code_hash: code_hash,
+			initializer_params_hash: None,
+		}));
+	});
+}
+
+#[test]
+fn upgrade_as_signed_fails() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::signed(AccountId32::new([0; 32]));
+		let address: H160 = Default::default();
+		let code_hash: H256 = Default::default();
+
+		assert_noop!(EthereumSystem::upgrade(origin, address, code_hash, None), BadOrigin);
+	});
+}
+
+#[test]
+fn upgrade_with_params() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let address: H160 = Default::default();
+		let code_hash: H256 = Default::default();
+		let initializer: Option<Initializer> =
+			Some(Initializer { params: [0; 256].into(), maximum_required_gas: 10000 });
+		assert_ok!(EthereumSystem::upgrade(origin, address, code_hash, initializer));
+	});
+}
+
+#[test]
+fn set_operating_mode() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let mode = OperatingMode::RejectingOutboundMessages;
+
+		assert_ok!(EthereumSystem::set_operating_mode(origin, mode));
+
+		System::assert_last_event(RuntimeEvent::EthereumSystem(crate::Event::SetOperatingMode {
+			mode,
+		}));
+	});
+}
+
+#[test]
+fn set_operating_mode_as_signed_fails() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::signed([14; 32].into());
+		let mode = OperatingMode::RejectingOutboundMessages;
+
+		assert_noop!(EthereumSystem::set_operating_mode(origin, mode), BadOrigin);
+	});
+}
+
+#[test]
+fn set_pricing_parameters() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let mut params = Parameters::get();
+		params.rewards.local = 7;
+
+		assert_ok!(EthereumSystem::set_pricing_parameters(origin, params));
+
+		assert_eq!(PricingParameters::<Test>::get().rewards.local, 7);
+	});
+}
+
+#[test]
+fn set_pricing_parameters_as_signed_fails() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::signed([14; 32].into());
+		let params = Parameters::get();
+
+		assert_noop!(EthereumSystem::set_pricing_parameters(origin, params), BadOrigin);
+	});
+}
+
+#[test]
+fn set_pricing_parameters_invalid() {
+	new_test_ext(true).execute_with(|| {
+		let origin = RuntimeOrigin::root();
+		let mut params = Parameters::get();
+		params.rewards.local = 0;
+
+		assert_noop!(
+			EthereumSystem::set_pricing_parameters(origin, params),
+			Error::<Test>::InvalidPricingParameters
+		);
```

Comment:
full coverage:
```suggestion
		
		let mut params = Parameters::get();
		params.exchange_rate = 0u128.into();
		assert_noop!(
			EthereumSystem::set_pricing_parameters(origin.clone(), params),
			Error::<Test>::InvalidPricingParameters
		);

		params = Parameters::get();
		params.fee_per_gas = sp_core::U256::zero();
		assert_noop!(
			EthereumSystem::set_pricing_parameters(origin.clone(), params),
			Error::<Test>::InvalidPricingParameters
		);

		params = Parameters::get();
		params.rewards.local = 0;
		assert_noop!(
			EthereumSystem::set_pricing_parameters(origin.clone(), params),
			Error::<Test>::InvalidPricingParameters
		);

		params = Parameters::get();
		params.rewards.remote = sp_core::U256::zero();
		assert_noop!(
			EthereumSystem::set_pricing_parameters(origin, params),
			Error::<Test>::InvalidPricingParameters
		);
```

Reply:
Thanks for the extra coverage, added here: https://github.com/paritytech/polkadot-sdk/pull/2522/commits/b49e11abac04a68e5178870a14d175618ac8f83e#diff-7f7d47672a96a1c5c4bb72df1bbdd352415500d235f1683e820ffa3847b5fedcR180